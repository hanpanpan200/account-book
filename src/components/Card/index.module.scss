.container {
  border-radius: 5px;
  margin: 10px;
  box-shadow: 0 6px 16px -6px #a4a4a4;

  .title {
    height: 26px;
    font-size: 16px;
    font-weight: 400;
    line-height: 26px;
    padding: 10px;
    background-color: #f5f5f5;
  }

  .content {
    padding: 10px;
  }
}