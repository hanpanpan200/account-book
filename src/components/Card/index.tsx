import React from 'react';

import styles from './index.module.scss';

interface Props {
  title: string
  children: JSX.Element | JSX.Element[]
}

const Card:React.FC<Props> = ({title, children}) => {
  return (
    <div className={styles.container}>
      <div className={styles.title}>{title}</div>
      <div className={styles.content}>
        {children}
      </div>
    </div>
  )
}

export default Card;