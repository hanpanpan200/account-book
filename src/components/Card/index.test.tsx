import React from 'react';
import { render } from '@testing-library/react';
import Card from './index';

describe('Card', () => {
  it('should render correctly when multiple items are provided', () => {
    const { container } = render(
      <Card title='卡片标题'>
        <div>Item1</div>
        <div>Item2</div>
      </Card>
    );
    expect(container).toMatchSnapshot();
  });
  it('should render correctly when only one item is provided', () => {
    const { container } = render(
      <Card title='卡片标题'>
        <div>Item1</div>
      </Card>
    );
    expect(container).toMatchSnapshot();
  });
});

