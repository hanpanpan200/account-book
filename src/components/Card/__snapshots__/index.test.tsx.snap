// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly when multiple items are provided 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      卡片标题
    </div>
    <div
      class="content"
    >
      <div>
        Item1
      </div>
      <div>
        Item2
      </div>
    </div>
  </div>
</div>
`;

exports[`Card should render correctly when only one item is provided 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      卡片标题
    </div>
    <div
      class="content"
    >
      <div>
        Item1
      </div>
    </div>
  </div>
</div>
`;
